### DISCUSSION TITLE
Supabase Local Development Storage?

### DISCUSSION THREAD

**Discussion by masnwilliams** [supabase/supabase#7230] [Questions] [answered]
Hello! I already have my local development set up using the Supabase CLI but am having difficulties with using Storage. I've read [this blurb](https://supabase.com/blog/2021/07/27/storage-beta#using-storage-in-local-development) and the links it recommends but can't tell if it automatically is set up or I have to do any extra steps to get file storage working.

Any help is greatly appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
There is by default a local file system setup so storage will work in local development without needing to setup an s3 bucket.  
But, Studio does not currently support storage UI, so you have to set up the RLS on the bucket and object tables with SQL.
This shows setting up RLS with SQL https://supabase.com/docs/guides/storage

**Reply by masnwilliams** [reply]
Ok, thank you! Would be great to get it working with the UI but this will work for me.

**Comment by abhay187** [comment]
@GaryAustin1 any update on this? Where can I see the URL of the file stored in storage?

**Reply by abhay187** [reply]
It's been more than 6 months. I hope you guys will release the storage UI soon.

**Reply by GaryAustin1** [reply]
I thought I heard they released it for the local dev side last launch week, but I don't do local development and a quick search did not reveal a statement, so not sure.

**Reply by barbinbrad** [reply]
I see storage on my localhost after upgrading to the newest supabase CLI:

![Screen Shot 2022-12-25 at 7 37 52 PM](https://user-images.githubusercontent.com/64510427/209488720-56b8c5ef-b212-479b-a600-a72d907a6be3.jpg)

**Reply by abhay187** [reply]
Yes, I can see it after updating. Thanks Supabase Team

**Reply by yasz** [reply]
but is that working?

**Reply by mb21** [reply]
Does anybody have the localhost URL I need to supply to tus-js-client when adapting [this example from the official docs](https://supabase.com/docs/guides/storage/uploads/resumable-uploads) to localhost? What's the equivalent to `https://${SUPABASE_PROJECT_ID}.supabase.co/storage/v1/upload/resumable` on localhost? `http://localhost:54323/storage/v1/upload/resumable` gives a CORS error, so probably not the right endpoint.
Thanks!

**Reply by rudolphmax** [reply]
@mb21 use `$ supabase status` to get the storage endpoint. What you're using is the studio URL, I believe. For me, it's `http://127.0.0.1:54321/storage/v1/s3`. So, I use `http://127.0.0.1:54321/storage/v1/upload/resumable` for TUS, for example. Notice that I removed `/s3` from the URL.
